app/core/security.py

```
@@ -145,26 +148,73 @@ def create_access_token(data: dict, expires_delta: Optional[timedelta] = None) -
+            logger.warning("[AUTH] Token has expired")
             raise HTTPException(
                 status_code=status.HTTP_401_UNAUTHORIZED,
                 detail="Token has expired",
                 headers={"WWW-Authenticate": "Bearer"},
             )
-    except (JWTError, ValidationError):
+        logger.info(f"[AUTH] Local JWT validated successfully for user: {token_data.sub}")
+        return token_data
+    except (JWTError, ValidationError) as e:
+        logger.error(f"[AUTH] JWT validation error: {e}")
         raise credentials_exception
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)